lib/oli_web/live_session_plugs/require_enrollment.ex

```diff
@@ -13,14 +15,26 @@ defmodule OliWeb.LiveSessionPlugs.RequireEnrollment do
         _session,
         %{
           assigns: %{
-            current_user: user,
-            section: %Sections.Section{requires_enrollment: false} = section
+            current_user: %User{} = user,
+            section: %Section{requires_enrollment: false} = section
           }
         } = socket
-      )
-      when not is_nil(user) do
-    if user, do: Sections.enroll(user.id, section.id, [ContextRoles.get_role(:context_learner)])
-    {:cont, assign(socket, is_enrolled: true)}
+      ) do
+    if Sections.has_enrollment?(user.id, section.slug) do
+      {:cont, assign(socket, is_enrolled: true)}
+    else
+      with :ok <- Sections.ensure_direct_delivery_enrollment_allowed(user, section) do
```

Comment:
It seems like the current logic for `ensure_direct_delivery_enrollment_allowed` is going to allow DD users to enroll in an LTI section:

```  
def ensure_direct_delivery_enrollment_allowed(_user, %Section{open_and_free: false}), do: :ok

  def ensure_direct_delivery_enrollment_allowed(nil, %Section{}), do: :ok

  def ensure_direct_delivery_enrollment_allowed(%User{} = user, %Section{open_and_free: true}) do
    if user.independent_learner do
      :ok
    else
      {:error, :non_independent_user}
    end
  end
```